test/topology_custom/test_mv_backlog.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+import requests
+import re
+
+logger = logging.getLogger(__name__)
+
+async def wait_for_view(cql, name, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' AND view_name = '{name}' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+def get_metrics(addr):
+    # The Prometheus API is on port 9180, and always http
+    addr = 'http://' + addr + ':9180/metrics'
+    resp = requests.get(addr)
+    if resp.status_code != 200:
+        pytest.skip('Metrics port 9180 is not available')
+    response = requests.get(addr)
+    assert response.status_code == 200
+    return response.text
+
+def get_metric(metrics, name, requested_labels=None):
+    the_metrics = get_metrics(metrics)
+    total = 0.0
+    lines = re.compile('^'+name+'{.*$', re.MULTILINE)
+    for match in re.findall(lines, the_metrics):
+        a = match.split()
+        metric = a[0]
+        val = float(a[1])
+        # Check if match also matches the requested labels
+        if requested_labels:
+            # we know metric begins with name{ and ends with } - the labels
+            # are what we have between those
+            got_labels = metric[len(name)+1:-1].split(',')
+            # Check that every one of the requested labels is in got_labels:
+            for k, v in requested_labels.items():
+                if not f'{k}="{v}"' in got_labels:
+                    # No match for requested label, skip this metric (python
+                    # doesn't have "continue 2" so let's just set val to 0...
+                    val = 0
+                    break
+        total += float(val)
+    return total
+
+def get_replicas(cql, key):
+    return cql.cluster.metadata.get_replicas("ks", Int32Type.serialize(key, cql.cluster.protocol_version))
+
+async def enable_on_servers(manager, servers, injection, oneshot=False):
+    errs = [manager.api.enable_injection(s.ip_addr, injection, oneshot) for s in servers]
+    return await asyncio.gather(*errs)
```

Comment:
It looks like we have this utility in `test_topology_failure_recovery.py` already (`inject_error_on`). Please move your utility function to `test/pylib/util.py` and make `test_topology_failure_recovery.py` use it (also, `inject_error_on` is a better name IMO because it's more descriptive, so please use it).

Reply:
My two cents: please move utility functions to "util.py" after you make sure they are useful and general, not on the first time you write them for the needs of one particular test. When you have several pieces of code that need the same utility, you have the perspective what features they all need and what is NOT general enough for several tests.

A good bad example is get_tablet_replicas() which someone (Tomek?) used in one test, so he quickly put it in test/pylib/tablets.py. But he created an implementation that only works for base tables, not for views. The second test that needed this function (test/topology_experimental_raft/test_mv_tablets.py) needed it for views, so I had to come up with a different function that works correctly both for tables and for views. This "make everything a utility function" approach is how in the dtest repository we have, after several years, a bazillion utility functions each with 20 different parameters which a very hard to read or write. I don't want to see us repeat this mistake. Code duplication is easier to fix (by de-duplication) than reuse of badly-designed code just for the sake of reuse.

Reply:
After all, I removed this method altogether, so it will need to wait for another PR that reimplements it to be moved to util.py